arguments.ts

```diff
@@ -38,11 +38,31 @@ async function parseArguments(args: MultiPointer, options: ParseArgument): Promi
   return [argObject]
 }
 
+const argumentPropPattern = new RegExp(`^${ns.code('argument#').value}(.+)$`)
+
 async function parseArgument(arg: AnyPointer, { context, variables, basePath, loaderRegistry }: ParseArgument): Promise<unknown> {
   if (!isGraphPointer(arg)) {
     throw new Error('Cannot load argument. Expected a single node or RDF List.')
   }
 
```

Comment:
```suggestion
  const keyValuePairs = []
  for (const quad of arg.dataset.match(arg.term)) {
    const isArgumentProp = quad.predicate.value.match(argumentPropPattern)
    if (isArgumentProp) {
      const key = isArgumentProp[1]
      const value = await parseArgument(clownface({ dataset: arg.dataset, term: quad.object }), { context, variables, basePath, loaderRegistry })
      keyValuePairs.push([key, value])
    }
  }
```
as much as I like functional programming, I still find more readable a good old imperative loop, especially when there is some `async` around

Reply:
I hear you. Awaiting the loop is smelly but admittedly, `reduce` is not much of an improvement. Let me refactor do use `Promise.all()` instead

Reply:
with `map` it looks much more straightforward to me. I keep thinking that `reduce` (and fold operators in general) is a powerful tool but somewhat low-level